From 2fcbf12c759c351ecf865b1b7a0d4a3dda48d48f Mon Sep 17 00:00:00 2001
From: Benedikt Spranger <b.spranger@linutronix.de>
Date: Fri, 29 Jul 2016 11:40:54 +0200
Subject: [PATCH 2/9] wireshark: glib compile fix

libvirt wireshark dissector uses wireshark and glib. Since wireshark
development files did not depend on glib add an expicit dependency.

Signed-off-by: Benedikt Spranger <b.spranger@linutronix.de>
---
 m4/virt-wireshark.m4 | 2 ++
 tools/Makefile.am    | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/m4/virt-wireshark.m4 b/m4/virt-wireshark.m4
index 2eee8a6ce8..90cd3c4a77 100644
--- a/m4/virt-wireshark.m4
+++ b/m4/virt-wireshark.m4
@@ -26,6 +26,7 @@ AC_DEFUN([LIBVIRT_ARG_WIRESHARK],[
 
 AC_DEFUN([LIBVIRT_CHECK_WIRESHARK],[
   LIBVIRT_CHECK_PKG([WIRESHARK_DISSECTOR], [wireshark], [2.4.0])
+  LIBVIRT_CHECK_PKG([GLIB], [glib-2.0], [2.46.2])
 
   dnl Check for system location of wireshark plugins
   if test "x$with_wireshark_dissector" != "xno" ; then
@@ -68,4 +69,5 @@ AC_DEFUN([LIBVIRT_CHECK_WIRESHARK],[
 
 AC_DEFUN([LIBVIRT_RESULT_WIRESHARK],[
   LIBVIRT_RESULT_LIB([WIRESHARK_DISSECTOR])
+  LIBVIRT_RESULT_LIB([GLIB])
 ])
diff --git a/tools/Makefile.am b/tools/Makefile.am
index 29fdbfe846..09ec45ac4f 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -418,8 +418,8 @@ if WITH_WIRESHARK_DISSECTOR
 
 ws_plugin_LTLIBRARIES = wireshark/src/libvirt.la
 wireshark_src_libvirt_la_CFLAGS = \
-	-I wireshark/src $(WIRESHARK_DISSECTOR_CFLAGS) $(XDR_CFLAGS)
-wireshark_src_libvirt_la_LDFLAGS = -avoid-version -module
+	-I wireshark/src $(WIRESHARK_DISSECTOR_CFLAGS) $(XDR_CFLAGS) $(GLIB_CFLAGS)
+wireshark_src_libvirt_la_LDFLAGS = -avoid-version -module $(GLIB_LIBS)
 wireshark_src_libvirt_la_SOURCES = \
 	wireshark/src/packet-libvirt.h \
 	wireshark/src/packet-libvirt.c \
-- 
2.23.0

